### PR TITLE
Issue 2132 - Fetch Balance object on sorting accounts (Explorer Page)

### LINKED ISSUE #2132: [1.5] Sorting Explorer Accounts by Balance breaks page
**Describe the bug**
The sorting for BalanceComponent in Explore -> Accounts doesn't work as expected, after few clicks appears white screen.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to Explore
2. Click on Accounts
3. Click on Current Balance column several times

**Expected behaviour**
Sort BalanceComponent without white screen.

### PR DESCRIPTION
**Pull request by startailcoon**
<h2>General</h2>
Closes #2132 

- Fixed loading of Balance Object on sorting accounts on explorer page
- Caching object result to reduce calls

<h2>General</h2>
lease make sure the following is done:

- Pull request is onto develop
- If you haven't already, have a look at 
  - https://github.com/bitshares/bitshares-ui/blob/develop/CODE_OF_CONDUCT.md
  - https://github.com/bitshares/bitshares-ui/blob/develop/CONTRIBUTING.md

<h2>Code Preparation</h2>

_Please review all your changes one last time before committing_

- [x] Check for unused code
- [x] No unrelated changes are included
- [x] None of the changed files are reformatting only
- [x] Code is self explanatory or documented
- [x] All written text is properly translated (english language)

<h2>Testing</h2>

_The branch has been tested on the following browsers (desktop and mobile view)_

- [ ] Chrome 
- [ ] Opera
- [x] Firefox
- [ ] Safari

<h2>User interface changes</h2>

_Delete this section if there weren't any UI changes. Please make sure you tested your changes in all themes_

- [x] Dark
- [x] Light
- [x] Midnight

_Please provide screenshots/licecap of your changes below_

### CODE REVIEW

File: app/components/Explorer/Accounts.jsx [hunk 1]
```diff
@@ -177,14 +193,29 @@ class Accounts extends React.Component {
             },
             {
                 title: <Translate component="span" content="account.percent" />,
-                dataIndex: "accountPercentages",
-                key: "accountPercentages",
+                dataIndex: "accountBalance",
```
Comment (sschiessl-bcp) [inline app/components/Explorer/Accounts.jsx]: was this on purpose, or mistaken?

File: app/components/Explorer/Accounts.jsx [hunk 2]
```diff
@@ -177,14 +193,29 @@ class Accounts extends React.Component {
...
+                key: "accountBalancePercentage",
                 sorter: (a, b) => {
-                    a.accountPercentages = parseFloat(a.accountPercentages);
-                    b.accountPercentages = parseFloat(b.accountPercentages);
-                    return a.accountPercentages > b.accountPercentages
+                    let a_balance = 0;
```
Comment (sschiessl-bcp) [inline app/components/Explorer/Accounts.jsx]: same logic as above? remove code duplication.

Also, are you certain that `accountPercentages` and `accountBalance` will always result in same sorting?

Comment (startailcoon) [reply app/components/Explorer/Accounts.jsx]: Redundant code fixed. 

`accountPercentages` and `accountBalance` is always the same value, the balance object ID, as seen here https://github.com/bitshares/bitshares-ui/pull/2518/files/43ed20ff1751f6980eb17ce6b8bb6acb3da5cd90#diff-db5bfe202a0f1034f9c94ffab4f37ebaL245

File: app/components/Explorer/Accounts.jsx [hunk 3]
```diff
@@ -73,6 +74,20 @@ class Accounts extends React.Component {
...
+    _getBalanceObject(object_id) {
+        let {balanceObjects} = this.state;
+
+        if(object_id && typeof(object_id) === "string") {
+            if(!balanceObjects[object_id]) {
+                balanceObjects[object_id] = parseFloat(ChainStore.getObject(object_id).get("balance"));
+            }
+        }
+
+        this.setState({balanceObjects});
```
Comment (sschiessl-bcp) [inline app/components/Explorer/Accounts.jsx]: Is it on purpose that the cache is in the state of this componenet? Will cause heacy re-renderings, won't it?

Comment (startailcoon) [reply app/components/Explorer/Accounts.jsx]: Perhaps, changed to not use setState for the cache.

File: app/components/Explorer/Accounts.jsx (hunk 1 above)
Comment (startailcoon) [reply app/components/Explorer/Accounts.jsx]: Purpose, see next comment

**Pull request merged**